backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -149,6 +149,12 @@ export class FeatureFlagService {
     return updatedState;
   }
 
+  public async updateFilterMode(flagId: string, filterMode: FILTER_MODE): Promise<FeatureFlag> {
+    // TODO: Add log for updating filter mode
+    const updatedFilterMode = await this.featureFlagRepository.updateFilterMode(flagId, filterMode);
```

Comment:
This should be wrapped in try catch block with proper http error code might be 
404 - in case feature flag doesn't exist
other wise 500.